packages/calypso-e2e/src/element-helper.ts

```diff
@@ -153,6 +153,50 @@ export async function getIdFromBlock( block: Locator ): Promise< string > {
 	return blockId;
 }
 
+/**
+ * Waits until DOM mutations for given target element become idle. Can be used
+ * in situations where Playwright auto-waiting is not working for some reason.
+ *
+ * @example
+ * await Promise.all( [
+ *   waitForMutations( page, '.foobars-wrapper' ),
+ *   page.click( 'button.load-foobars' ),
+ * ] );
+ * const foobarsText = await page.innerText( '.foobars' );
+ * @param {Page} page Page object
+ * @param {string} selector Observer target selector
+ * @param {object} options Observer options
+ */
+export async function waitForMutations(
+	page: Page | Frame,
+	selector: string,
+	options = {
+		attributes: true,
+		subtree: true,
+		childList: true,
+	}
+): Promise< void > {
+	const target = await page.waitForSelector( selector );
+
+	await page.evaluate(
+		async ( args ) => {
+			await new Promise( ( resolve ) => {
+				const debounceResolve = () => {
+					let timer: NodeJS.Timeout;
+					return () => {
+						clearTimeout( timer );
+						timer = setTimeout( resolve, 1000 );
```

Comment:
**Suggestion (non-blocking)**: Can expose the milliseconds here as a function parameter? (I think it can have a default of `1000`).

I love this helper so much. ❤️  To answer you question in the PR, I think this will have _huge_ utility across E2E testing Calypso! But I think we might want to make how long we wait for the mutations to be configurable to get the max utility. Most mutations happen quickly, but I feel like there are some (like ones that depend on a web request response) that might be slower.

Reply:
> I think we might want to make how long we wait for the mutations to be configurable to get the max utility. Most mutations happen quickly, but I feel like there are some (like ones that depend on a web request response) that might be slower.

The 1000ms here is actually the debounce timeout, not the general timeout for the method. The difference is that the timer will start only when the first mutation kicks in. For example, when triggered before the click in a `Promise.all` call, it will wait until the mutations under the given parent start and then timeout if no more mutations happen after 1000ms:

```js
await Promise.all( [
  waitForMutations( page, '.images' ),
  page.click( '.load-images' ),
] );
```

Having said that, we might want a general timeout for this method anyway. If there are no mutations for some reason, it will wait for the global timeout, which is way too much for the case. Let's say 10 seconds is more than enough to timeout this one! - Will update in a bit.

Reply:
Updated in https://github.com/Automattic/wp-calypso/pull/69345/commits/053d8dd2b6990527f161d7889910c3969d80c30d.

Reply:
@dpasque, could you take another look? 🙏

Reply:
@WunderBart Looks great to me! 😄